app/src/main/java/org/mozilla/fenix/components/BackgroundServices.kt

```diff
@@ -101,28 +101,29 @@ class BackgroundServices(
             // Notify observers for subscription changes.
             it.registerForSubscriptions(object : PushSubscriptionObserver {
                 override fun onSubscriptionAvailable(subscription: AutoPushSubscription) {
-                    accountManager.authenticatedAccount()?.deviceConstellation()
-                        ?.setDevicePushSubscriptionAsync(
-                            DevicePushSubscription(
-                                endpoint = subscription.endpoint,
-                                publicKey = subscription.publicKey,
-                                authKey = subscription.authKey
+                    // Update for only the services subscription.
+                    if (subscription.type == PushType.Services) {
+                        accountManager.authenticatedAccount()?.deviceConstellation()
+                            ?.setDevicePushSubscriptionAsync(
```

Comment:
The `subscribeForType` call below will result in `onSubscriptionAvailable` being called. So you'd call this method on every startup. Underneath, it's expected to modify an FxA device record. Seems like you'll write to FxA servers on every app startup here, which isn't something we can land as-is.

Reply:
I've updated the PR to remove the token from prefs and force registration only once. We can remove this in https://github.com/mozilla-mobile/fenix/issues/4218 when majority of our users are on the latest nightly or release.